findimports.py

```diff
@@ -7,27 +7,37 @@
 Syntax: findimports.py [options] [filename|dirname ...]
 
 Options:
-  -h, --help        This help message
-
-  -i, --imports     Print dependency graph (default action).
-  -d, --dot         Print dependency graph in dot (graphviz) format.
-  -n, --names       Print dependency graph with all imported names.
-
-  -u, --unused      Print unused imports.
-  -a, --all         Print unused imports even if there's a comment.
-  --duplicate       Print duplicate imports.
-  -v                Print more information (only affects --duplicate).
-
-  -N, --noext       Omit external dependencies.
-
-  -p, --packages    Convert the module graph to a package graph.
-  -l N, --level N   Collapse subpackages deeper than the Nth level.
-
-  -c, --collapse    Collapse dependency cycles.
-  -T, --tests       Collapse packages named 'tests' and 'ftests' with parent
-                    packages.
-
-FindImports requires Python 2.7 or later.
+  -h, --help            Show this help message and exit.
+
+  -i, --imports         Print dependency graph (default action).
+  -d, --dot             Print dependency graph in dot (graphviz) format.
+  -n, --names           Print dependency graph with all imported names.
+
+  -u, --unused          Print unused imports.
+  -a, --all             Don't ignore unused imports when there's a comment on
+                        the same line (only affects -u).
+  --duplicate           Warn about duplicate imports.
+  -v, --verbose         Print more information (currently only 
+                        affects --duplicate).
+
+  -N, --noext           Omit external dependencies.
+
+  -p, --packages        Convert the module graph to a package graph.
+  -l N, --level=N       Collapse subpackages to the topmost Nth levels.
+                 
+  -c, --collapse        Collapse dependency cycles.
+  -T, --tests           Collapse packages named 'tests' and 'ftests' with
+                        parent packages.
+
+  -w FILE, --write-cache=FILE
+                        Write a pickle cache of parsed imports; provide the
+                        cache filename as the only non-option argument to load
+                        it back.
+  -I FILE, --ignore=FILE
+                        Ignore a file or directory; this option can be used
+                        multiple times. Default: ['venv']
+
+FindImports requires Python 3.6 or later.
```

Comment:
Oh wow, and I'd totally forgotten that there was a usage message in the docstring!  Thank you for noticing and updating it!

(This code is _old_, these days I'd rely on argparse-generated help messages, and maybe put a copy in the README for new user convenience, and then usually forget to update it when making changes.)

Reply:
I'm making modules too so I know the feeling when I forget to update docs 😅